src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -265,6 +265,8 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                     _packageRestoreManager.PackageRestoredEvent -= PackageRestoreManager_PackageRestored;
                     _packageRestoreManager.PackageRestoreFailedEvent -= PackageRestoreManager_PackageRestoreFailedEvent;
 
+                    _packageRestoreManager.RaiseAssetsFileMissingEventForProjectAsync(false);
```

Comment:
So I actually noticed an interesting thing: _restoreEventsPublisher.OnSolutionRestoreCompleted(
                            new SolutionRestoredEventArgs(_status, solutionDirectory)); is some event that's being raised. 
                            
                          Does it have the same properties as our event? 
                          Basically it says, restore has happened, which in essence is what this one does?
                          
                          Worth checking if it's equivalent.

Reply:
I still have to learn more about events, but after looking at this, `OnSolutionRestoreCompleted` is used to save the status (`NuGetOperationStatus`) of the last restore. I think is not equivalent, can be useful but it's not 100% clear to me how because we only want to change the banner in a successful status. Do you think it's worth spend more time investigating this?

Reply:
> is used to save the status (NuGetOperationStatus) of the last restore

The content of the event is not super relevant though right? And it could be tweaked. 

I was only bringing that up just so that we can maybe avoid new public APIs. 

I'm comfortable with whatever you decide.